src/shared/featureController.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SettingsConfiguration } from './settingsConfiguration'
+
+/**
+ * This class handles feature access for unreleased features.
+ * Access can be granted to a feature by adding a boolean flag to the VS Code settings.json file.
+ * Features can be designated as session-permanent: this means that the setting won't change if altered mid-session
+ * To designate a feature as session-permanent, add the feature to the this.permanentSettings array.
+ *
+ * Example: use the following setting to add a feature called `newFeature1`:
+ * {
+ * ...
+ *     "aws.toggle.newFeature1": true,
+ * ...
+ * }
+ * This feature can then be accessed by calling FeatureController.isFeatureActive('newFeature1')
+ */
+export class FeatureController {
+
+    private readonly permanentSettingsMap: Map<string, boolean>
+
+    // This array holds keys for session-permanent settings.
+    // These settings are set as permanent at extension load.
+    // Changing these keys' values in settings.json during the extension's lifecycle
+    //     will not reflect until the next time the extension loads.
+    private readonly permanentSettings: string[] = [
```

Comment:
This field only appears to be used in the constructor. Does it need to be a class member?

Reply:
I was going to ask for some feedback on this, I'm not even sure if this is the best way to maintain this (so that way, if we want to alter the session-permanent features, we don't have to touch the class itself). I'm debating having it as a separate array/set/JSON object outside of the class/file that's imported on init.

Reply:
Moved array outside the class ✔️